### PR TITLE
Add a watchdog for peer shutdown (avoids deadlock if any of protocols is blocked)

### PR DESCRIPTION
**Pull request by mandrigin**

### CODE REVIEW

File: p2p/peer.go [hunk 1]
```diff
@@ -240,7 +240,26 @@ loop:
...
+	/* A watchdog for the waitGroup (p.wg)
+	* When a peer shuts down we give 5 seconds for all protocols to finish,
+	* but we continue the process anyway (and print a warning message).
+	* Otherwise, this peer will be stuck in the list of peers (`admin_peers`)
+	* and won't be able to reconnect until the app is restarted.
+	* */
+	c := make(chan struct{})
+
+	go func() {
+		defer close(c)
```
Comment (dshulyak) [inline p2p/peer.go]: i don't think that defer makes sense here

Comment (mandrigin) [reply p2p/peer.go]: yeah, you are right, will remove it

Comment (mandrigin) [reply p2p/peer.go]: done

File: p2p/peer.go [hunk 2]
```diff
@@ -240,10 +240,37 @@ loop:
...
+	c := make(chan struct{})
+
+	go func() {
+		p.wg.Wait()
+		close(c)
+	}()
+
+	select {
+	case <-c:
+		p.Log().Trace("Peer stopped successfully")
```
Comment (adambabik) [inline p2p/peer.go]: Maybe `Debug` instead of `Trace`?

Comment (mandrigin) [reply p2p/peer.go]: makes sense, will fix

**Pull request merged**